app/models/open_weather_map.rb

```diff
@@ -0,0 +1,47 @@
+COOLDOWN_IN_MINUTES = 10
+OPEN_WEATHER_MAP_URI = 'http://api.openweathermap.org/'
+OPEN_WEATHER_MAP_PATH = 'data/2.5/forecast/daily'
+OPEN_WEATHER_MAP_CRITERIA = 'zip=22630,us'
+OPEN_WEATHER_MAP_KEY = '3fcb4e23443750cc1256f3016a160ca7'
+OPEN_WEATHER_MAP_FULL_URL = OPEN_WEATHER_MAP_URI + OPEN_WEATHER_MAP_PATH + '?' + OPEN_WEATHER_MAP_CRITERIA + '&appid=' + OPEN_WEATHER_MAP_KEY
+
+class OpenWeatherMap < ApplicationRecord
+  validates :last_get, presence: true
+
+  def current_weather_payload
+    if OpenWeatherMap.safe_to_poll(DateTime.now)
+      self.get_weather
+    else
+      self.retrieve_cache
+    end
+    self.response
+  end
+
+  def get_weather
+    api_response = HTTParty.get(OPEN_WEATHER_MAP_FULL_URL)
+    if api_response.success?
+      self.response = api_response.body
+      self.api_key = OPEN_WEATHER_MAP_KEY
```

Comment:
I'm over-engineering this at present (since I only have the one key), but I was imagining that I might have users provide their own keys once/if I support multiple users. I also thought I might want to store the key in the database rather than code. Will remove.

Here's an example of what I'll be parsing:
```
{"city":{"id":0,"name":"Front Royal","coord":{"lon":-78.1796,"lat":38.9268},"country":"US","population":0},"cod":"200","message":9.7998518,"cnt":7,"list":[{"dt":1502211600,"temp":{"day":298.9,"min":287.51,"max":299.4,"night":287.51,"eve":296.13,"morn":298.9},"pressure":1005.15,"humidity":67,"weather":[{"id":801,"main":"Clouds","description":"few clouds","icon":"02d"}],"speed":2.36,"deg":340,"clouds":24},{"dt":1502298000,"temp":{"day":298.46,"min":287.16,"max":299,"night":287.16,"eve":295.24,"morn":290.38},"pressure":1010.22,"humidity":60,"weather":[{"id":800,"main":"Clear","description":"sky is clear","icon":"02d"}],"speed":1.75,"deg":119,"clouds":8},{"dt":1502384400,"temp":{"day":299.48,"min":287.74,"max":299.91,"night":287.74,"eve":295.93,"morn":289.8},"pressure":1008.67,"humidity":52,"weather":[{"id":801,"main":"Clouds","description":"few clouds","icon":"02d"}],"speed":1.77,"deg":66,"clouds":24},{"dt":1502470800,"temp":{"day":295.77,"min":291.91,"max":295.77,"night":291.91,"eve":294.57,"morn":292.67},"pressure":1006.62,"humidity":0,"weather":[{"id":501,"main":"Rain","description":"moderate rain","icon":"10d"}],"speed":1.15,"deg":117,"clouds":85,"rain":5.98},{"dt":1502557200,"temp":{"day":300.18,"min":291.84,"max":300.18,"night":294.06,"eve":296.1,"morn":291.84},"pressure":1002.65,"humidity":0,"weather":[{"id":501,"main":"Rain","description":"moderate rain","icon":"10d"}],"speed":2.8,"deg":166,"clouds":8,"rain":6.22},{"dt":1502643600,"temp":{"day":299.08,"min":294.71,"max":299.08,"night":294.71,"eve":297.68,"morn":295.22},"pressure":1000.01,"humidity":0,"weather":[{"id":501,"main":"Rain","description":"moderate rain","icon":"10d"}],"speed":2.21,"deg":213,"clouds":29,"rain":7.18},{"dt":1502730000,"temp":{"day":298.98,"min":291.26,"max":298.98,"night":291.26,"eve":296.13,"morn":295.28},"pressure":1000.06,"humidity":0,"weather":[{"id":500,"main":"Rain","description":"light rain","icon":"10d"}],"speed":1.61,"deg":2,"clouds":85,"rain":2.56}]}
```